be/src/main/java/com/jjikmuk/sikdorak/review/service/ReviewService.java

```diff
@@ -14,22 +16,23 @@
 public class ReviewService {
 
 	private final StoreService storeService;
+	private final UserService userService;
 	private final ReviewRepository reviewRepository;
 
 	@Transactional
-	public Long insertReview(ReviewInsertRequest request) {
-		Long findStoreId = request.getStoreId();
-		storeService.findById(findStoreId);
+	public Review insertReview(LoginUser loginUser, ReviewInsertRequest reviewInsertRequest) {
+		userService.searchById(loginUser.getId());
+		storeService.searchById(reviewInsertRequest.getStoreId());
```

Comment:
p2;
여기 검색 로직은 id 가 존재하는지 검증하는 부분이군요.
단순히 검색 로직만 있으니, 검증하는 부분이라는 맥락이 잘 보이지 않는 것 같습니다.
이 부분이 userId 와 storeId 를 검증한다는 의미를 잘 전달할 수 있게 바뀌면 좋을 것 같습니다!

Reply:
저도 네이밍이 아쉬웠지만 위와 같이 구현한 이유는 다음과 같아요.


만약 service.isExistingById 메서드로 존재 여부를 구분한다면

ReviewService 서비스에서


```java
// 기존코드

		userService.searchById(loginUser.getId());
		storeService.searchById(reviewInsertRequest.getStoreId());


// 변경 코드
                if (storeService.isExistingById(..)) {
                  throw new NotFoundStore()
                }
                
                if (userService.isExistingById(..)) {
                  throw new NotFoundUser()
                }
```


으로 두 번 if 조건문이 들어가고, 또한 리뷰서비스에서  스토어, 유저 예외를 던져야하는 어려움이 있어서요.

유효성 검증을  각 서비스의 search 메서드에서 하는 것으로 구현했어요

---

혹시 다른 아이디어 있으신가요?

Reply:
제이가 말씀해주신 내용도 공감이 되네요.
그렇다고 null을 반환하고 검증을 밖에서 하게되면 해당 로직이 너무 중복되는 작업이 될것 같은데..
차라리 validateXXX 메서드를 만들어서 해당 메서드 내에서 예외를 던지도록 하고 search는 이름 그대로 검색의 역할만 가져가도 좋을거 같아요!

Reply:
```suggestion
		userService.searchByIdOrThrow(loginUser.getId());
		storeService.searchByIdOrThrow(reviewInsertRequest.getStoreId());
```

두분 말씀을 듣고보니 외부에서 처리하려니 번거로운 것 같아서, 이런 형태로 네이밍 해보는건 어떨까요?

Reply:
채널로 토론

<img width="399" alt="image" src="https://user-images.githubusercontent.com/57086195/184305803-0b67ba8d-b182-46cd-9b35-8172b96fa97b.png">

Reply:
reference

https://parkadd.tistory.com/13

https://medium.com/webeveloper/%EC%8A%A4%ED%94%84%EB%A7%81-%EC%88%9C%ED%99%98-%EC%B0%B8%EC%A1%B0-circular-reference-d01c6beee7e6

https://velog.io/@sumusb/Spring-Service-Layer%EC%97%90-%EB%8C%80%ED%95%9C-%EA%B3%A0%EC%B0%B0